### ISSUE TITLE
Autodoc failing to import *some* modules, only on RTD

### ISSUE THREAD

**Issue by anicusan** [readthedocs/readthedocs.org#8622]
## Details
Hello, I have a rather odd situation where essentially the same Sphinx conf / code I used for another Python library ([pept](https://pept.readthedocs.io/en/latest/)) works perfectly, but for [Coexist](https://coexist.readthedocs.io/en/latest/) produces warnings *only* on readthedocs.

My `.readthedocs.yml` file is quite vanilla:
```
version: 2

sphinx:
  configuration: docs/source/conf.py

formats: all

python:
  install:
    - requirements: requirements.txt
    - requirements: requirements_extra.txt
    - method: pip
      path: .
  system_packages: true
```

* Read the Docs project URL: https://coexist.readthedocs.io/en/latest/
* Build URL (if applicable): https://readthedocs.org/api/v2/build/15104962.txt
* Read the Docs username (if applicable): anicusan

## Expected Result

Creating the docs **locally** works perfectly, with no warnings produced. I tested on new, empty environments with Python 3.7 and 3.9; the library is written in pure Python, no C extensions or deeply-nested module hierarchies. I expected correct autosummaries on RTD too.

## Actual Result

Looking at the build logs, the library installation went fine; there are no import errors: `Successfully installed coexist-0.1.0`. However, when generating the docstrings with autosummary & numpydoc, I get multiple warnings *only on RTD*, e.g. `WARNING: autosummary: failed to import coexist.plots.access`; however, other objects at the same nesting level **are** found, e.g. `writing output... [ 44%] manual/generated/coexist.schedulers.Scheduler`.

In summary:

- Locally everything works fine, even in fresh environments.
- On RTD the library is installed correctly, and *some* library functions are found by autosummary.
- However, many other functions get `autosummary: failed to import`, even at the same nesting level.

No amount of fiddling, Googling and reading documentation seemed to solve this.
Any help would be *greatly* appreciated.

Best wishes,
Leonard
